src/main/java/io/cdap/plugin/record/splitter/RecordSplitterConfig.java

```diff
@@ -0,0 +1,169 @@
+ */
+public class RecordSplitterConfig extends PluginConfig {
+  public static final String FIELD_TO_SPLIT = "fieldToSplit";
+  public static final String DELIMITER = "delimiter";
+  public static final String OUTPUT_FIELD = "outputField";
+  public static final String SCHEMA = "schema";
+
+  @Name(FIELD_TO_SPLIT)
+  @Description("Specifies the field to split.")
+  @Macro
+  private String fieldToSplit;
+
+  @Name(DELIMITER)
+  @Description("Specifies the delimiter used to split each record. If using escape characters, " +
+    "be sure to double escape. So \n = \\n")
+  @Macro
+  private String delimiter;
+
+  @Name(OUTPUT_FIELD)
+  @Description("Specifies the name of the output field.")
+  @Macro
+  private String outputField;
+
+  @Name(SCHEMA)
+  @Description("Specifies the schema that has to be output.")
+  private String schema;
+
+  public RecordSplitterConfig(String fieldToSplit, String delimiter, String outputField, String schema) {
+    this.fieldToSplit = fieldToSplit;
+    this.delimiter = delimiter;
+    this.outputField = outputField;
+    this.schema = schema;
+  }
+
+  private RecordSplitterConfig(Builder builder) {
+    fieldToSplit = builder.fieldToSplit;
+    delimiter = builder.delimiter;
+    outputField = builder.outputField;
+    schema = builder.schema;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(RecordSplitterConfig copy) {
+    return builder()
+      .setFieldToSplit(copy.fieldToSplit)
+      .setDelimiter(copy.delimiter)
+      .setOutputField(copy.outputField)
+      .setSchema(copy.schema);
+  }
+
+  public String getFieldToSplit() {
+    return fieldToSplit;
+  }
+
+  public String getDelimiter() {
+    return delimiter;
+  }
+
+  public String getOutputField() {
+    return outputField;
+  }
+
+  public Schema getSchema() {
+    try {
+      return Schema.parseJson(schema);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Unable to parse output schema.", e);
+    }
+  }
+
+  public void validate(FailureCollector failureCollector, Schema inputSchema) {
+    try {
+      Schema outSchema = Schema.parseJson(schema);
+      if (!containsMacro(OUTPUT_FIELD) && outSchema.getField(outputField) == null) {
+        failureCollector.addFailure("Output Schema must contain the specified output field.", null)
+          .withConfigProperty(OUTPUT_FIELD);
+      }
+    } catch (IOException e) {
+      failureCollector.addFailure("Unable to parse output schema.", null)
+        .withStacktrace(e.getStackTrace())
+        .withConfigProperty(SCHEMA);
+      return;
+    }
+    if (!containsMacro(FIELD_TO_SPLIT)) {
+      Schema inputFieldSchema = inputSchema.getField(fieldToSplit).getSchema();
+
+      Schema.Type inputFieldType = inputFieldSchema.isNullable()
+        ? inputFieldSchema.getNonNullable().getType()
+        : inputFieldSchema.getType();
+      if (inputFieldType != Schema.Type.STRING) {
```

Comment:
I am not sure if logical types can be string. But probably would be best to check that this is not a logical type as per guidelines.